contracts/utils/Arrays.sol

```diff
@@ -0,0 +1,43 @@
+pragma solidity ^0.4.23;
+
+import "../math/Math.sol";
+
+
+/**
+ * @title Arrays
+ * @dev Utility library of inline array functions
+ */
+library Arrays {
+
+  /**
+   * @dev Find upper bound for searching element. Utilize binary search function with O(log(n)) cost.
```

Comment:
We should improve this documentation: in particular, it is critical to document that `element` is the value we're searching for, that _array_ is sorted in ascending order, and the return value is an _index_.

We should also better explain what the function does: given said sorted array, it returns the smallest array index such that the value at that index is larger or equal to the received element, or `array.length` if no such index exists (i.e. all values inside the array are smaller that the target). It also returns 0 for an empty array, though we may want to revert in that scenario.